src/scripts/main.js

```diff
@@ -1,3 +1,578 @@
+        && arrayCells[i - 1][j] !== 0
+        && arrayCells[i][j] === arrayCells[i - 1][j]
+        && arrayCells[i - 1][j] !== undefined
+        && merger[i] === true
+        && merger[i - 1] === true
+      ) {
+        arrayCells[i][j] += arrayCells[i][j];
+        arrayCells[i - 1][j] = 0;
+        merger[i] = false;
+        score += arrayCells[i][j];
+        continue;
+      }
+    }
+
+    for (let i = 0; i < arrayCells.length; i++) {
+      // переміщення по пустих значеннях
+      if (
+        i > 0
+        && arrayCells[i][j] === 0
+        && arrayCells[i - 1][j] !== 0
+        && arrayCells[i - 1][j] !== undefined
+      ) {
+        arrayCells[i][j] = arrayCells[i - 1][j];
+        arrayCells[i - 1][j] = 0;
+        continue;
+      }
+    }
+  }
+}
+
+function moveRight() {
+  for (let i = 0; i < arrayCells.length; i++) {
+    const merger = [true, true, true, true];
+
+    for (let j = 0; j < arrayCells[i].length; j++) {
+      // переміщення по пустих значеннях
+      if (
+        arrayCells[i][j] === 0
+        && arrayCells[i][j - 1] !== 0
+        && arrayCells[i][j - 1] !== undefined
+      ) {
+        arrayCells[i][j] = arrayCells[i][j - 1];
+        arrayCells[i][j - 1] = 0;
+        continue;
+      }
+    }
+
+    for (let j = 0; j < arrayCells[i].length; j++) {
+      if (// Для переміщення трьох нерівних
+        arrayCells[i][j + 1] !== 0
+        && arrayCells[i][j + 2] !== 0
+        && arrayCells[i][j] !== 0
+        && arrayCells[i][j + 3] === 0
+      ) {
+        arrayCells[i][j + 3] = arrayCells[i][j + 2];
+        arrayCells[i][j + 2] = arrayCells[i][j + 1];
+        arrayCells[i][j + 1] = arrayCells[i][j];
+        arrayCells[i][j] = 0;
+        continue;
+      }
+
+      if ( // Для переміщення двох нерівних
+        arrayCells[i][j - 1] !== 0
+        && arrayCells[i][j - 1] !== undefined
+        && arrayCells[i][j] !== arrayCells[i][j - 1]
+        && arrayCells[i][j] !== 0
+        && arrayCells[i][j + 1] === 0
+      ) {
+        arrayCells[i][j + 1] = arrayCells[i][j];
+        arrayCells[i][j] = arrayCells[i][j - 1];
+        arrayCells[i][j - 1] = 0;
+        continue;
+      }
+
+      //
+
+      if ( // Для додавання
+        arrayCells[i][j - 1] !== 0
+        && arrayCells[i][j] === arrayCells[i][j - 1]
+        && arrayCells[i][j - 1] !== undefined
+        && merger[j] === true
+        && merger[j - 1] === true
+      ) {
+        arrayCells[i][j] += arrayCells[i][j];
+        arrayCells[i][j - 1] = 0;
+        merger[j] = false;
+        score += arrayCells[i][j];
+        continue;
+      }
+    }
+
+    for (let j = 0; j < arrayCells[i].length; j++) {
+      // Переміщення по пустих значеннях
+      if (
+        arrayCells[i][j] === 0
+        && arrayCells[i][j - 1] !== 0
+        && arrayCells[i][j - 1] !== undefined
+      ) {
+        arrayCells[i][j] = arrayCells[i][j - 1];
+        arrayCells[i][j - 1] = 0;
+        continue;
+      }
+    }
+  }
+};
+
+function moveLeft() {
```

Comment:
The function 'moveLeft' has complex nested loops and conditionals. Consider breaking it down into smaller functions for better readability and maintainability.